dashboard/test/lib/expired_deleted_account_purger_test.rb

```diff
@@ -0,0 +1,263 @@
+require 'test_helper'
+require 'expired_deleted_account_purger'
+
+class ExpiredDeletedAccountPurgerTest < ActiveSupport::TestCase
+  freeze_time
+
+  setup_all do
+    # No soft-deleted users and no purged users, to begin with.
+    assert_empty User.with_deleted.where(purged_at: nil).where.not(deleted_at: nil)
+    assert_empty User.with_deleted.where.not(purged_at: nil)
```

Comment:
One thing I'm worried about - parts of this test check the number of deleted and purged users found for metrics purposes, and I'm not sure our environments will consistently be clean when we get to this test.  I've added this setup_all step so there's an obvious failure when this goes wrong, but I may need to find a nicer way to isolate this test while still checking correct behavior.

Reply:
How about, for the specific tests that rely on the number for metrics, stub out the query to always return a set number of mock users? And for all other tests, assert each expected user is in the set, but don't compare the whole set in case there's other data in the environment (which it looks like we're already doing)?